src/lib/controller/http-client.ts

```diff
@@ -0,0 +1,107 @@
+import { PUBLIC_API_BASE_URL } from "$env/static/public";
+
+/**
+ * A simple HTTP client that can make requests to the backend API.
+ *
+ * This class is used to make requests to the backend API. It is a simple wrapper around the `fetch` API.
+ * It uses the `PUBLIC_API_BASE_URL` environment variable to determine the base URL of the API.
+ *
+ * @example
+ * const client = new HttpClient("api/v1/example");
+ * await client.get();
+ * await client.post({ data: "example" });
+ * await client.put({ data: "example" });
+ * await client.patch({ data: "example" });
+ * await client.delete();
+ */
+export class HttpClient {
+    private readonly basePath: string;
+
+    /**
+     * Creates a new HTTP client.
+     *
+     * @param path The path to the API endpoint. This path will be appended to the `PUBLIC_API_BASE_URL` environment variable.
+     *
+     * @example
+     * const client = new HttpClient("api/v1/example");
```

Comment:
```suggestion
     * const client = new HttpClient("/api/v1/example");
```

or am I wrong (for your current implementation)?

Reply:
In the current implementation, the slash is added for a given path.

Reply:
My mistake.